### PR TITLE
feat: add ICurveStableSwapNG

### LINKED ISSUE #931: Pairing Dollar with 3CRV will still make Dollar follow a price of $1.03, even if StableSwapMetaNG pools are used
Pairing Dollar with the tripool will still cause [this issue](https://github.com/sherlock-audit/2023-12-ubiquity-judging/issues/15), even if the pool used is a StableSwapMetaNG and not the old curve pools. 
Recommendation:
The best solution to achieve an actual peg to $1 would be to use Chainlink price feeds instead. or to pair Dollar with a different asset that is actually pegged to $1 rather than 3CRV, given that currently the pool price will tend to follow 3CRV's price, which currently trades at $1.03.

_Originally posted by @0xadrii in https://github.com/ubiquity/ubiquity-dollar/pull/893#discussion_r1572711362_

### PR DESCRIPTION
**Pull request by rndquu**
Resolves https://github.com/ubiquity/ubiquity-dollar/issues/931

This PR refactors [LibUbiquityPool](https://github.com/ubiquity/ubiquity-dollar/blob/81f760bdbcf8c9b467ba55014f78ae86c5ca68de/packages/contracts/src/dollar/libraries/LibUbiquityPool.sol) to use [ICurveStableSwapNG](https://docs.curve.fi/stableswap-exchange/stableswap-ng/pools/plainpool/) instead of [ICurveStableSwapMetaNG](https://docs.curve.fi/stableswap-exchange/stableswap-ng/pools/metapool/).

`ICurveStableSwapMetaNG` is basically a `Dollar/3CRV` pool. Keeping in mind that `3CRV` is equal to `$1.03` we can't reliably fetch Dollar price in USD from `ICurveStableSwapMetaNG`.

We plan to create a new `ICurveStableSwapNG` pool with `LUSD/Dollar` pair. So calculating Dollar price in USD works this way:
1) Fetch `LUSD/USD` price from chainlink
2) Fetch `Dollar/LUSD` price from Curve's `ICurveStableSwapNG` [oracle](https://docs.curve.fi/stableswap-exchange/stableswap-ng/pools/oracles/)
3) Now we have everything to calculate `Dollar/USD` price

P.S. Deployment scripts will be updated in a separate PR

### CODE REVIEW

**Comment by gitcoindev** [comment]
Implementation-wise looks perfect, I will wait for the security research with his review before approval to merge.

**Comment by 0xadrii** [comment]
Everything looks correct, the issue is properly fixed

**Review by gitcoindev — APPROVED** [review]

**Review by molecula451 — APPROVED** [review]
Indeed, great work as usual, rndquu

**Pull request merged**
